### PR TITLE
PMM-7517: Backup artifacts list API

### PR DESCRIPTION
**Pull request by oter**
https://jira.percona.com/browse/PMM-7517

### CODE REVIEW

**Review by artemgavrilov — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]

**Review by Dasio — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]
🏇

**Review by palash25 — APPROVED** [review]

File: api/managementpb/backup/backups.proto [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+// DataModel is a model used for performing a backup.
+enum DataModel {
+  DATA_MODEL_INVALID = 0;
+  PHYSICAL = 1;
+  LOGICAL = 2;
+}
+
+// Backup represents single Backup Location.
+message Backup {
```
Comment (atymchuk) [inline api/managementpb/backup/backups.proto:28]: @oter What about backup type (incremental vs full)? Or do you mean we add it later?

File: api/managementpb/backup/backups.proto [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+message ListBackupsResponse {
+  repeated Backup backups = 1;
+}
+
+// Backups service provides public methods for managing Backups.
+service Backups {
+  // ListBackups returns a list of all backups.
+  rpc ListBackups(ListBackupsRequest) returns (ListBackupsResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/backup/Backups/List"
```
Comment (atymchuk) [inline api/managementpb/backup/backups.proto:62]: Any reason why the signature is different from what I see on other /management endpoint?
https://pmmdemo.percona.com/swagger

Comment (atymchuk) [reply api/managementpb/backup/backups.proto:62]: If we follow the same approach as here https://pmmdemo.percona.com/swagger#/SecurityChecks/ListSecurityChecks, then it should probably be `/v1/management/Backups/List`

File: api/swagger/swagger-dev.json [hunk 3]
```diff
@@ -21438,6 +21438,150 @@
         }
       }
     },
+    "/v1/management/backup/Backups/List": {
```
Comment (atymchuk) [inline api/swagger/swagger-dev.json:21441]: The url should be consistent with the rest of the management APIs, PTAL https://pmmdemo.percona.com/swagger#/

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: api/managementpb/backup/backups.proto (hunk 1 above)
Comment (oter) [reply api/managementpb/backup/backups.proto:28]: we will add it later

File: api/managementpb/backup/backups.proto (hunk 2 above)
Comment (oter) [reply api/managementpb/backup/backups.proto:62]: inspired by https://github.com/percona/pmm/blob/release/2.16/api/managementpb/ia/channels.proto#L176

Comment (atymchuk) [reply api/managementpb/backup/backups.proto:62]: Ok, I see your point! But then it should probably mean "inventory", like this `post: "/v1/management/backup/Inventory/List"`. Pls check p.3.1.1 in https://confluence.percona.com/display/PMM/Backup+Management. And also the design below, to stay consistent.
BackupManagement - Inventory

**Review by artemgavrilov — APPROVED** [review]
LGTM, but routes renaming requested by Alex sounds reasonable to me.

File: api/managementpb/backup/backups.proto (hunk 2 above)
Comment (oter) [reply api/managementpb/backup/backups.proto]: as per discussion renamed to `v1/management/backup/Artifacts/List`

File: api/managementpb/backup/json/backup.json [hunk 4]
```diff
@@ -15,6 +15,150 @@
...
+                        "type": "string",
+                        "x-order": 5
+                      },
+                      "service_name": {
+                        "description": "Service name.",
+                        "type": "string",
+                        "x-order": 6
+                      },
+                      "status": {
+                        "description": "Status shows current status of Backup.",
```
Comment (atymchuk) [inline api/managementpb/backup/json/backup.json:95]:
```suggestion
                        "description": "Status shows current status of backup.",
```

File: api/managementpb/backup/json/backup.json [hunk 5]
```diff
@@ -15,6 +15,150 @@
...
+                        "x-order": 3
+                      },
+                      "location_name": {
+                        "description": "Location name.",
+                        "type": "string",
+                        "x-order": 4
+                      },
+                      "name": {
+                        "type": "string",
+                        "title": "Artifact name",
```
Comment (atymchuk) [inline api/managementpb/backup/json/backup.json:81]:
```suggestion
                        "title": "Backup artifact name",
```

**Review by Dasio — APPROVED** [review]

File: api/managementpb/backup/json/backup.json [hunk 6]
```diff
@@ -15,6 +15,150 @@
...
+        "responses": {
+          "200": {
+            "description": "A successful response.",
+            "schema": {
+              "type": "object",
+              "properties": {
+                "artifacts": {
+                  "type": "array",
+                  "items": {
+                    "description": "Artifact represents single Backup Artifact.",
```
Comment (atymchuk) [inline api/managementpb/backup/json/backup.json:44]:
```suggestion
                    "description": "Artifact represents single backup artifact.",
```

**Review by atymchuk — APPROVED** [review]
All my comments are nits, feel free to merge.

**Pull request merged**